sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -128,7 +128,8 @@ object PartitioningUtils {
       //   "hdfs://host:9000/invalidPath"
       //   "hdfs://host:9000/path"
       // TODO: Selective case sensitivity.
-      val discoveredBasePaths = optDiscoveredBasePaths.flatten.map(_.toString.toLowerCase())
+      val discoveredBasePaths =
+      optDiscoveredBasePaths.flatten.map(_.toString.toLowerCase(Locale.ROOT))
```

Comment:
I am worried of this one too. It sounds the path could contains Turkish characters I guess..